tests/libs/common/common_tests.cpp

```diff
@@ -155,7 +155,15 @@ ring_buffer_test_event_handler(_Inout_ void* ctx, _In_opt_ const void* data, siz
     auto records = event_context->records;
     auto it = std::find(records->begin(), records->end(), event_record);
     if (it != records->end()) {
-        event_context->matched_entry_count++;
+        auto index = std::distance(records->begin(), it);
+        if (event_context->event_received.find(index) != event_context->event_received.end()) {
+            // Duplicate event record received.
+            // Return error so that no further callback is made.
+            return -1;
+        } else {
+            event_context->event_received.insert(index);
```

Comment:
alternatively, remove the matched entry from records ?

Reply:
Awesome suggestion! That makes it a lot simpler.

Reply:
Actually, that doesn't work. The function bindmonitor_ring_buffer_test assumes the list of records is immutable.